### DISCUSSION TITLE
[Bug]: How to convert model.ckpt files in the lightning directory to .onnx or openvino files

### DISCUSSION THREAD

**Discussion by fat-921** [open-edge-platform/anomalib#1324] [Q&A] [answered]
### Describe the bug

I want to convert the model.ckpt file in lightning directory to onnx or openvino format separately and save metadata.json at the same time. Is there any convenient way to convert?
I don't want to retrain and then turn the model...

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

None

### OS information

OS information:
- OS: Windows 10
- Python version: [e.g. 3.8.10]
- Anomalib version: [e.g. 0.3.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: 1x GeForce RTX 3060
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

Can you give me sample code?

### Pip/GitHub

pip

### Configuration YAML

```yaml
any
```

### Logs

```shell
None
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fat-921** [comment]
Here is my code. the model that comes out of this code cannot detect anything

```
import logging
from argparse import ArgumentParser, Namespace
from anomalib.config import get_configurable_parameters
from anomalib.data import get_datamodule
from anomalib.models import get_model

logger = logging.getLogger("anomalib")
from anomalib.deploy import TorchInferencer, export, ExportMode
from pathlib import Path

def get_parser() -> ArgumentParser:
    """Get parser.

    Returns:
        ArgumentParser: The parser object.
    """
    parser = ArgumentParser()
    parser.add_argument("--config", type=str, required=False, help="Path to a model config file")
    parser.add_argument("--weights", type=str, required=True, help="Path to model weights")
    parser.add_argument("--infer_size", type=str, default=[256, 256], required=False,
                        help="infer image size(height, width), like 224 or [224,224] or 224,224")
    args = parser.parse_args()

    # 如果宽高为整数,转换为数组
    if isinstance(args.infer_size, int):
        args.infer_size = [args.infer_size, args.infer_size]
    # tuple转换为list
    if isinstance(args.infer_size, tuple):
        args.infer_size = list(args.infer_size)
    return args


def export_model() -> None:
    """将训练完的模型.ckpt转成onnx后在转成engine
    Args:
        args: The arguments from the command line.
    Returns:
    """
    args = get_parser()
    config = get_configurable_parameters(config_path=args.config, weight_file=args.weights)

    datamodule = get_datamodule(config)
    model = get_model(config)
    
    Path(config.project.path).mkdir(parents=True, exist_ok=True)
    dirpath = config.project.path
    
    export(
        task=datamodule.test_data.task,
        input_size=args.infer_size,
        transform=datamodule.test_data.transform.to_dict(),
        model=model,
        export_root=dirpath,
        export_mode=ExportMode.ONNX,
    )
    
    
if __name__ == "__main__":
    export_model()
```

**Comment by blaz-r** [comment]
Exporting is part of the training, but in this case, where you don't want to re-train, I'm not sure what exactly would be the solution. Your code looks fine, but I think that the model doesn't have the weight initialized. If you look inside [get_model](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/models/__init__.py#L137-L138) you can see that state dict is loaded from **relative path from project root** saved in `config.init_weights`, which you don't have initialized as far as I can see. 

Additionally, are you using provided OpenVINO inferencer, because there are still pre-processing and post-processing steps involved in successful inference.

**Comment by fat-921** [comment]
I just found that the model.ckpt file in the `lightning folder` is only the initial saved file, not the ckpt file saved after training, so its inference result is empty. If I want to save this ckpt file after training, where should I change it?

**Comment by blaz-r** [comment, marked as answer]
model.ckpt in lighting should be the one saved after training. You can load it in the way I said in the comment above, or you should add 
```    
model = model.load_from_checkpoint("path to model.ckpt")
``` 
to your code. Withe the current code you are not loading the weights at all.

**Comment by fat-921** [comment]
> model.ckpt in lighting should be the one saved after training. You can load it in the way I said in the comment above, or you should add
> 
> ```
> model = model.load_from_checkpoint("path to model.ckpt")
> ```
> 
> to your code. Withe the current code you are not loading the weights at all.

Ok, thank you. I just realized that I didn't load weights
